src/platform_linux/shmem.c

```diff
@@ -39,35 +80,26 @@ static bool Trace_large_frags  = FALSE;
  *       We track both for improved debugging.
  *
  * Lifecyle:
+ *  - Initially all frag_addr will be NULL => tracking fragment is empty
  *  - When a large fragment is initially allocated, frag_addr / frag_size will
  *    be set.
  *  - (allocated_to_pid != 0) && (freed_by_pid == 0) - Fragment is in use.
  *  - (allocated_to_pid != 0) && (freed_by_pid != 0) - Fragment is free.
  * ---------------------------------------------------------------------------
  */
+// clang-format off
 typedef struct shm_frag_info {
-   void *shm_frag_addr;                // Start address of this memory fragment
-                                       // NULL => tracking fragment is empty
-   size_t   shm_frag_size;             // bytes
-   int      shm_frag_allocated_to_pid; // Allocated to this OS-pid
-   threadid shm_frag_allocated_to_tid; // Allocated to this Splinter thread-ID
-   int      shm_frag_freed_by_pid;     // OS-pid that freed this
-   threadid shm_frag_freed_by_tid;     // Splinter thread-ID that freed this
+   void       *frag_addr;             // Start addr of this memory fragment
+   const char *frag_func;             // Calling func which lead to creation
+   size_t      frag_size;             // bytes
+   threadid    frag_allocated_to_tid; // Alloc'ed to this Splinter thread-ID
+   int         frag_allocated_to_pid; // Allocated to this OS-pid
+   int         frag_freed_by_pid;     // OS-pid that freed this fragment
+   threadid    frag_freed_by_tid;     // Splinter thread-ID that freed this
+   int         frag_line;
 } shm_frag_info;
 
-/*
- * All memory allocations of this size or larger will be tracked in the
- * above fragment tracker array. For large inserts workload, we allocate large
- * memory chunks for fingerprint array, which is more than a MiB. For scans,
- * splinterdb_iterator_init() allocates memory for an iterator which is ~92+KiB.
- * Set this to a lower value so we can re-cycle free fragments for iterators
- * also.
- */
-#if SPLINTER_DEBUG
-#   define SHM_LARGE_FRAG_SIZE (90 * KiB)
-#else
-#   define SHM_LARGE_FRAG_SIZE (38 * KiB)
-#endif // SPLINTER_DEBUG
```

Comment:
Diff in this limit was causing unit-tests to fail in debug builds and pass in release build.

This artifact is leftover from my poc-dev days ... no longer should be needed to separate out limits.

Have stabilized on 32K as lower limit for large fragments in both builds. Moved to `shmem.h`